### PR TITLE
hbase support batch

### PR DESCRIPTION
**Pull request by stuBirdFly**
hbase support batch

## Summary

hbase support batch,support multiple CFs in a batch and allow both put and delete operations within the same batch.

### CODE REVIEW

File: src/main/java/com/alipay/oceanbase/hbase/OHTable.java [hunk 1]
```diff
@@ -1677,36 +1631,73 @@ private KeyValue modifyQualifier(KeyValue original, byte[] newQualifier) {
...
+                        singleOpResultNum++;
+                        KeyValue kv = new KeyValue(delete.getRow(), delete.getTimeStamp(),
+                            KeyValue.Type.Maximum);
+                        batch.addOperation(buildMutation(kv, DEL, isTableGroup));
+                    } else {
+                        for (Map.Entry<byte[], List<KeyValue>> entry : delete.getFamilyMap()
+                            .entrySet()) {
+                            byte[] family = entry.getKey();
+                            List<KeyValue> keyValueList = entry.getValue();
+                            List<com.alipay.oceanbase.rpc.mutation.Mutation> mutations = new LinkedList<>();
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/hbase/OHTable.java:1677]: why not use import `com.alipay.oceanbase.rpc.mutation.Mutation` and use Mutation directly

File: src/main/java/com/alipay/oceanbase/hbase/OHTable.java [hunk 2]
```diff
@@ -1677,36 +1631,73 @@ private KeyValue modifyQualifier(KeyValue original, byte[] newQualifier) {
...
+                            for (KeyValue kv : keyValueList) {
+                                com.alipay.oceanbase.rpc.mutation.Mutation mutation;
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/hbase/OHTable.java:1679]: same as above

File: src/main/java/com/alipay/oceanbase/hbase/OHTable.java [hunk 3]
```diff
@@ -1677,36 +1631,73 @@ private KeyValue modifyQualifier(KeyValue original, byte[] newQualifier) {
...
+                                if(isTableGroup){
+                                    KeyValue new_kv = modifyQualifier(kv,(Bytes.toString(family) + "." + Bytes.toString(kv.getQualifier()))
+                                            .getBytes());
+                                    mutation = buildMutation(new_kv, DEL, isTableGroup);
+                                } else {
+                                    mutation = buildMutation(kv, DEL, isTableGroup);
+                                }
+                                if (KeyValue.Type.codeToType(kv.getTypeByte()) == KeyValue.Type.DeleteFamily){
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/hbase/OHTable.java:1687]: it's seems wrong in same cases, for example, we have a delete with two sub-operation, operation 1 is delete family with ts t1, and operation 2 is delete family with ts t2 and t1>t2, as a result, only  cells with ts <= t2 is deleted.

File: src/main/java/com/alipay/oceanbase/hbase/OHTable.java [hunk 4]
```diff
@@ -428,13 +429,63 @@ public Boolean[] exists(List<Get> gets) throws IOException {
     }
 
     @Override
-    public void batch(List<? extends Row> actions, Object[] results) {
-        throw new FeatureNotSupportedException("not supported yet.");
+    public void batch(final List<? extends Row> actions, final Object[] results) throws IOException {
```
[the remark this answers is not in the document]
Comment (stuBirdFly) [reply src/main/java/com/alipay/oceanbase/hbase/OHTable.java:432]: already defend in batch interface

File: src/main/java/com/alipay/oceanbase/hbase/OHTable.java [hunk 5]
```diff
@@ -428,13 +429,63 @@ public Boolean[] exists(List<Get> gets) throws IOException {
...
+        BatchError batchError = new BatchError();
+        try {
+            List<Integer> resultMapSingleOp = new LinkedList<>();
+            String realTableName = getTargetTableName(actions);
+            obTableClient.setRuntimeBatchExecutor(executePool);
+            BatchOperation batch = buildBatchOperation(realTableName, actions, tableNameString.equals(realTableName), resultMapSingleOp);
+            BatchOperationResult tmpResults = batch.execute();
+            if (results != null) {
+                if (results.length != actions.size()) {
+                    throw new AssertionError("results.length");
```
[the remark this answers is not in the document]
Comment (stuBirdFly) [reply src/main/java/com/alipay/oceanbase/hbase/OHTable.java:442]: same with hbase1.3.6

**Comment by shenyunlong** [comment]
LGTM

**Pull request merged**